src/test/java/io/jenkins/plugins/sprp/EndToEndTest.java

```diff
@@ -0,0 +1,119 @@
+package io.jenkins.plugins.sprp;
+
+import org.apache.tools.ant.filters.StringInputStream;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+
+import static org.junit.Assert.assertEquals;
+
+public class EndToEndTest {
+    @Rule
+    public JenkinsRule jenkinsRule = new JenkinsRule();
+
+    @Test
+    public void verifyIsUnix() {
+        boolean isUnixFormPipelineSnippetGenerator = new PipelineSnippetGenerator().isUnix();
+        boolean isUnixFormJenkinsRule = jenkinsRule.createLocalLauncher().isUnix();
+        assertEquals(isUnixFormPipelineSnippetGenerator, isUnixFormJenkinsRule);
+    }
+
+    @Test
+    public void yamlToPipelineTest() throws NoSuchMethodException, IOException, InstantiationException, IllegalAccessException, InvocationTargetException, NoSuchFieldException {
+        String yaml = "agent: any\n" +
```

Comment:
I'd suggest changing the name of the variables: something like `yaml` -> `inputYaml`, and `pipelineScriptLinux` -> `expectedPipelineLinux`